factor/totp/lang/en/factor_totp.php

```diff
@@ -30,6 +30,8 @@
 $string['devicename_help'] = 'This is the device you have an authenticator app installed on. You can setup multiple devices so this label helps track which ones are being used. You should setup each device with their own unique code so they can be revoked separately.';
 $string['error:wrongverification'] = 'Incorrect verification code';
 $string['error:codealreadyused'] = 'This code has already been used to authenticate. Please wait for a new code to be generated, and try again.';
+$string['error:oldcode'] = 'This code appears to be old. It may be out of date, or your authenticator device time may be out of sync. Please verify device time on your authenticator device, and try again.';
```

Comment:
I think to make things easier to debug can you echo the current server time + time zone in both of these